IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs

```diff
@@ -0,0 +1,251 @@
+                default:
+                    throw new ArgumentException($"Unknown browser: {browser}");
+            }
+
+            Driver.Manage().Window.Maximize();
+            Driver.Manage().Timeouts().PageLoad = TimeSpan.FromMinutes(2);
+        }
+
+        /// <summary>
+        /// Wraps the Selenium Driver's native web element to wait until the element exists before returning.
+        /// If you need to verify an element DOESN'T exist, then call Browser.Driver.FindElement directly.
+        /// </summary>
+        /// <param name="by">Selenium "By" statement to find the element</param>
+        /// <param name="secondsToWait">Seconds to wait while retrying before failing</param>
+        /// <returns></returns>
+		public WebElement FindElement(By by, int secondsToWait = 5)
+        {
+            Console.WriteLine($"Attempting to find element using selector: {by}");
+
+            // Eventually swap this out for our own wait
+            WebDriverWait wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(secondsToWait));
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+
+            return new WebElement(Driver.FindElement(by), by, Driver);
+        }
+
+        /// <summary>
+        /// Wraps the Selenium Driver's native web element to wait until at least one element exists before returning.
+        /// If you need to verify an element DOESN'T exist, then call Browser.Driver.FindElements directly.
+        /// </summary>
+        /// <param name="by">Selenium "By" statement to find the element</param>
+        /// <param name="secondsToWait">Seconds to wait while retrying before failing</param>
+        /// <returns></returns>
+        public ReadOnlyCollection<WebElement> FindElements(By by, int secondsToWait = 5)
+        {
+            Console.WriteLine($"Attempting to find all elements using selector: {by}");
+
+            // Eventually swap this out for our own wait
+            WebDriverWait wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(secondsToWait));
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(by));
+
+            return new ReadOnlyCollection<WebElement>(
+                            Driver.FindElements(by)
+                                    .Select(webElement => new WebElement(webElement, by, Driver))
+                                    .ToList());
+        }
+
+        /// <summary>
+        ///Waits until a function evaluates to true OR times out after a specified period of time
+        /// </summary>
+        /// <param name="func">Function to evaluate</param>
+        /// <param name="secondsToWait">Secondes to wait until timeout / return false</param>
+        /// <returns></returns>
+        public async Task<bool> WaitFor(Func<bool> func, int secondsToWait = 15)
+        {
+            DateTime end = DateTime.Now.AddSeconds(secondsToWait);
+            do
+            {
+                await Task.Delay(500);
```

Comment:
Yeah, in this particular case we were trying to eliminate the possibility of Selenium interfering with some JavaScript loading up some data on a webpage. Our thought process was if we pause a second before Selenium is invoked, it would give the JS a chance to finish executing. When I go through and refactor out a common retry method, I'll change the wait to happen after the function invocation.